src/main/java/com/es/phoneshop/model/cart/HttpSessionCartService.java

```diff
@@ -15,12 +16,14 @@
     private static final String CART_SESSION_ATTRIBUTE = "cart";
     private static HttpSessionCartService instance = new HttpSessionCartService();
     private ProductDao productDao;
-    private AddToCartResult addToCartResult;
+    private ProductService productService;
+    private CartServiceMethodsResult cartServiceMethodsResult;
     private static final String QUANTITY = "quantity";
 
     private HttpSessionCartService() {
         productDao = ArrayListProductDao.getInstance();
-        addToCartResult = new AddToCartResult();
+        productService = ProductService.getInstance();
+        cartServiceMethodsResult = new CartServiceMethodsResult();
```

Comment:
new instance must be created per operation

Reply:
did it